lib/ngx/re.lua

```diff
@@ -0,0 +1,159 @@
+-- I hereby assign copyright in this code to the lua-resty-core project,
+-- to be licensed under the same terms as the rest of the code.
+
+local ffi = require 'ffi'
+local bit = require "bit"
+local base = require "resty.core.base"
+local core_regex = require "resty.core.regex"
+
+
+local C = ffi.C
+local sub = string.sub
+local band = bit.band
+local new_tab = base.new_tab
+local tostring = tostring
+local re_match_compile = core_regex.re_match_compile
+local destroy_compiled_regex = core_regex.destroy_compiled_regex
+
+
+local PCRE_ERROR_NOMATCH = -1
+
+
+local FLAG_DFA           = 0x02
+
+
+ffi.cdef [[
```

Comment:
Definitions via `ffi.cdef` are global. So if you already require `resty.core.regex` then all the C symbols declared there should already be available.